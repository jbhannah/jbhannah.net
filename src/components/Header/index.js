import PropTypes from "prop-types"
import React from "react"
import github from "../../images/github.svg"
import keybase from "../../images/keybase.svg"
import linkedin from "../../images/linkedin.svg"
import twitter from "../../images/twitter.svg"
import { contentWidth, headerWidth, mq } from "../../utils/styles"
import Heading from "../Heading"
import Link from "../Link"

const Header = ({ title }) => (
  <header
    css={{
      display: "flex",
      flexDirection: "row",
      justifyContent: "space-between",
      margin: "0 auto",
      maxWidth: contentWidth,
      padding: "0 1rem",
      width: "100%",
      [mq.lg]: {
        flexDirection: "column",
        justifyContent: "flex-start",
        margin: 0,
        textAlign: "right",
        width: headerWidth,
      },
      [mq.xl]: {
        marginLeft: `calc(-1 * ${headerWidth})`,
      },
    }}
  >
    <Heading
      level="h1"
      css={{
        fontSize: "1.15rem",
        [mq.sm]: {
          fontSize: "1.25rem",
        },
      }}
    >
      <Link to="/">{title}</Link>
    </Heading>
    <ul
      css={{
        display: "flex",
        flexDirection: "row",
        justifyContent: "flex-end",
        listStyleType: "none",
        margin: 0,
        marginTop: "2.37rem",
        "& li": {
          float: "right",
          marginBottom: 0,
          marginRight: "0.5rem",
          width: "1.25rem",
          height: "1.25rem",
          "&:last-child": {
            marginRight: 0,
          },
          "& a": {
            background: "none",
            display: "block",
            height: "100%",
            width: "100%",
          },
          "& img": {
            transition: "border-radius 0.5s",
            borderRadius: "50%",
            display: "block",
            height: "100%",
            margin: 0,
            width: "100%",
          },
          "&:hover img": {
            borderRadius: "15%",
          },
        },
        [mq.lg]: {
          marginTop: 0,
          "& li": {
            height: "1.5rem",
            width: "1.5rem",
          },
        },
      }}
    >
      <li>
        <Link href="https://github.com/jbhannah" target="_blank">
          <img alt="jbhannah on GitHub" src={github} />
        </Link>
      </li>
      <li>
        <Link href="https://keybase.io/jbhannah" target="_blank">
          <img alt="jbhannah on Keybase" src={keybase} />
        </Link>
      </li>
      <li>
        <Link href="https://www.linkedin.com/in/jbhannah" target="_blank">
          <img alt="jbhannah on LinkedIn" src={linkedin} />
        </Link>
      </li>
      <li>
        <Link href="https://twitter.com/jbhannah" target="_blank">
          <img alt="@jbhannah on Twitter" src={twitter} />
        </Link>
      </li>
    </ul>
  </header>
)

Header.propTypes = {
  title: PropTypes.string,
}

export default Header
