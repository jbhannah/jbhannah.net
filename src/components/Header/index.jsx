import PropTypes from "prop-types"
import React from "react"
import { contentWidth, headerWidth, hoverRadius, mq } from "../../utils/styles"
import Heading from "../Heading"
import Link from "../Link"
import SocialNav from "../SocialNav"

const headerCSS = {
  display: "flex",
  flexDirection: "row",
  justifyContent: "space-between",
  margin: "0 auto",
  maxWidth: contentWidth,
  padding: "0 1rem",
  width: "100%",
  [mq.lg]: {
    flexDirection: "column",
    justifyContent: "flex-start",
    margin: 0,
    textAlign: "center",
    width: headerWidth,
  },
  [mq.xl]: {
    marginLeft: `-${headerWidth}`,
  },
}

const titleCSS = {
  fontSize: "1.15rem",
  [mq.sm]: {
    fontSize: "1.25rem",
  },
}

const avatarCSS = {
  display: "none",
  [mq.lg]: {
    display: "block",
    width: "100%",
    marginBottom: "0.79rem",
    "source, img": {
      ...hoverRadius,
      display: "block",
      margin: 0,
    },
  },
}

const Header = ({ title, avatar, socialLinks }) => (
  <header css={headerCSS}>
    <Heading level="h1" css={titleCSS}>
      <Link href="/" css={{ display: "block" }}>
        <picture css={avatarCSS}>
          <source type="image/webp" srcSet={avatar.srcSetWebp} />
          <source type="image/png" srcSet={avatar.srcSet} />
          <img src={avatar.src} alt={title} />
        </picture>
        {title}
      </Link>
    </Heading>
    <SocialNav {...{ socialLinks }} />
  </header>
)

export default Header

Header.propTypes = {
  title: PropTypes.string,
  avatar: PropTypes.shape({
    src: PropTypes.string,
    srcSet: PropTypes.string,
    srcSetWebp: PropTypes.string,
  }),
  socialLinks: SocialNav.propTypes.socialLinks,
}
