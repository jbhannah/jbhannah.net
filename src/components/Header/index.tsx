import { InterpolationWithTheme } from "@emotion/core"
import { graphql, useStaticQuery } from "gatsby"
import * as React from "react"
import { contentWidth, headerWidth, hoverRadius, mq } from "../../utils/styles"
import Heading from "../Heading"
import Link from "../Link"
import { SocialNav } from "../SocialNav"
import { HeaderQuery } from "./__generated__/HeaderQuery"

const headerCSS: InterpolationWithTheme<any> = {
  display: "flex",
  flexDirection: "row",
  justifyContent: "space-between",
  margin: "0 auto",
  maxWidth: contentWidth,
  padding: "0 1rem",
  width: "100%",
  [mq.lg]: {
    flexDirection: "column",
    justifyContent: "flex-start",
    margin: 0,
    textAlign: "center",
    width: headerWidth,
  },
  [mq.xl]: {
    marginLeft: `-${headerWidth}`,
  },
}

const titleCSS: InterpolationWithTheme<any> = {
  fontSize: "1.15rem",
  [mq.sm]: {
    fontSize: "1.25rem",
  },
}

const avatarCSS: InterpolationWithTheme<any> = {
  display: "none",
  [mq.lg]: {
    display: "block",
    width: "100%",
    marginBottom: "0.79rem",
    "source, img": {
      ...hoverRadius,
      display: "block",
      margin: 0,
    },
  },
}

export const Header = () => {
  const {
    site: {
      siteMetadata: { title },
    },
    file: {
      childImageSharp: { fixed: avatar },
    },
  } = useStaticQuery<HeaderQuery>(query)

  return (
    <header css={headerCSS}>
      <Heading level="h1" css={titleCSS}>
        <Link href="/" css={{ display: "block" }}>
          <picture css={avatarCSS}>
            <source type="image/webp" srcSet={avatar.srcSetWebp} />
            <source type="image/png" srcSet={avatar.srcSet} />
            <img src={avatar.src} alt={title} />
          </picture>
          {title}
        </Link>
      </Heading>
      <SocialNav />
    </header>
  )
}

const query = graphql`
  query HeaderQuery {
    site {
      siteMetadata {
        title
      }
    }
    file(sourceInstanceName: { eq: "images" }, name: { eq: "avatar" }) {
      childImageSharp {
        fixed(width: 160) {
          src
          srcSet
          srcSetWebp
        }
      }
    }
  }
`
