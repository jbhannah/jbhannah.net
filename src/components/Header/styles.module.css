@value layout: "../Layout/styles.module.css";
@value break-sm,
break-md,
break-lg,
break-xl,
content-width,
header-width from layout;

@custom-media --break-sm break-sm;
@custom-media --break-md break-md;
@custom-media --break-lg break-lg;
@custom-media --break-xl break-xl;

.header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 0 auto;
  max-width: content-width;
  padding: 0 1rem;
  width: 100%;
}

.title {
  font-size: 1.15rem;
}

.iconList {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  list-style-type: none;
  margin: 0;
  margin-top: 2.37rem;
}

.icon {
  float: right;
  margin-bottom: 0;
  margin-right: 0.5rem;
  width: 1.25em;
  height: 1.25rem;
}

.icon:last-child {
  margin-right: 0;
}

.icon a {
  background: none;
  display: block;
  height: 100%;
  width: 100%;
}

.icon img {
  transition: border-radius 0.5s;
  border-radius: 50%;
  display: block;
  height: 100%;
  margin: 0;
  width: 100%;
}

.icon:hover img {
  border-radius: 15%;
  /* match actual radius of icon corners */
}

@media (--break-sm) {
  .title {
    font-size: 1.25rem;
  }
}

@media (--break-lg) {
  .header {
    flex-direction: column;
    justify-content: flex-start;
    margin: 0;
    text-align: right;
    width: header-width;
  }

  .iconList {
    margin-top: 0;
  }

  .icon {
    height: 1.5rem;
    width: 1.5rem;
  }
}

@media (--break-xl) {
  .header {
    margin-left: calc(-1 * header-width);
  }
}
