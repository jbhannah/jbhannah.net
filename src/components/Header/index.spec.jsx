import { shallow } from "enzyme"
import React from "react"
import Header from "."

const TestHeader = props =>
  shallow(
    <Header
      title="Test Site"
      avatar={{
        src: "avatar-src.png",
        srcSet: "avatar-srcSet.png",
        srcSetWebp: "avatar-srcSet.webp",
      }}
      socialLinks={[
        {
          key: "Test Service",
          service: "Test Service",
          link: "https://www.example.com",
          name: "Test Name",
        },
      ]}
      {...props}
    />
  )

describe("Header", () => {
  const tree = TestHeader()

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot()
  })
})
