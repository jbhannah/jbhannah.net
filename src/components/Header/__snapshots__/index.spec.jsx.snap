// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  css={
    Object {
      "@media only screen and (min-width: 59rem)": Object {
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "margin": 0,
        "textAlign": "center",
        "width": "12rem",
      },
      "@media only screen and (min-width: 70rem)": Object {
        "marginLeft": "-12rem",
      },
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": "0 auto",
      "maxWidth": "42rem",
      "padding": "0 1rem",
      "width": "100%",
    }
  }
>
  <Heading
    css={
      Object {
        "@media only screen and (min-width: 28rem)": Object {
          "fontSize": "1.25rem",
        },
        "fontSize": "1.15rem",
      }
    }
    level="h1"
  >
    <Link
      css={
        Object {
          "display": "block",
        }
      }
      href="/"
    >
      <img
        css={
          Object {
            "@media only screen and (min-width: 59rem)": Object {
              "a:hover > &": Object {
                "borderRadius": "15%",
              },
              "borderRadius": "50%",
              "display": "block",
              "marginBottom": "0.79rem",
              "transition": "border-radius 0.5s",
              "width": "100%",
            },
            "display": "none",
          }
        }
        src="test-file-stub"
      />
      Test Site
    </Link>
  </Heading>
  <SocialNav
    socialLinks={
      Array [
        Object {
          "key": "Test Service",
          "link": "https://www.example.com",
          "name": "Test Name",
          "service": "Test Service",
        },
      ]
    }
  />
</header>
`;
